packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/create_transport.ts

```diff
@@ -41,7 +40,17 @@ export const createTransport = ({
 
     async request(params: TransportRequestParams, options?: TransportRequestOptions) {
       const opts: TransportRequestOptions = options ? { ...options } : {};
-      const opaqueId = getExecutionContext();
+      executionContext?.get()?.incrementCircuitBreaker('requests');
+      const requests = executionContext?.get()?.getCircuitBreaker('requests') ?? 0;
+      if (requests > 0 && requests % 100 === 0) {
+        console.log(
+          `CIRCUIT_BREAKER_ERROR: ${requests} requests generated by execution context ${executionContext
+            ?.get()
+            ?.toString()}`
+        );
+        // throw new Error('CIRCUIT_BREAKER_ERROR: too many requests');
```

Comment:
nit: should we create a new `CircuitBreakerError` so downstream consumers can react to it by stopping any further attempts or, in the case of Usage Collectors, decide whether to push the partial results they've collected?